File: src/main/java/org/mitallast/queue/raft/protocol/AppendSuccessful.java
package org.mitallast.queue.raft.protocol;

import org.mitallast.queue.common.stream.StreamInput;
import org.mitallast.queue.common.stream.StreamOutput;
import org.mitallast.queue.common.stream.Streamable;
import org.mitallast.queue.transport.DiscoveryNode;

import java.io.IOException;

public class AppendSuccessful implements Streamable {
    private final DiscoveryNode member;
    private final long term;
    private final long lastIndex;

    public AppendSuccessful(StreamInput stream) throws IOException {
        member = stream.readStreamable(DiscoveryNode::new);
        term = stream.readLong();
        lastIndex = stream.readLong();
    }

    public AppendSuccessful(DiscoveryNode member, long term, long lastIndex) {
        this.member = member;
        this.term = term;
        this.lastIndex = lastIndex;
    }

    public DiscoveryNode getMember() {
        return member;
    }

    public long getTerm() {
        return term;
    }

    public long getLastIndex() {
        return lastIndex;
    }

    @Override
    public void writeTo(StreamOutput stream) throws IOException {
        stream.writeStreamable(member);
        stream.writeLong(term);
        stream.writeLong(lastIndex);
    }

    @Override
    public String toString() {
        return "AppendSuccessful{" +
            "member=" + member +
            ", term=" + term +
            ", lastIndex=" + lastIndex +
            '}';
    }
}

File: src/main/java/org/mitallast/queue/raft/protocol/AppendSuccessful.kt
package org.mitallast.queue.raft.protocol

import org.mitallast.queue.common.codec.Codec
import org.mitallast.queue.common.codec.Message
import org.mitallast.queue.transport.DiscoveryNode

data class AppendSuccessful(val member: DiscoveryNode, val term: Long, val lastIndex: Long) : Message {
    companion object {
        val codec = Codec.of(
            ::AppendSuccessful,
            AppendSuccessful::member,
            AppendSuccessful::term,
            AppendSuccessful::lastIndex,
            DiscoveryNode.codec,
            Codec.longCodec(),
            Codec.longCodec()
        )
    }
}
